src/test/java/com/example/bistro/bistroapp/service/CustomerServiceImplTest.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bistro.bistroapp.service;
+
+import com.example.bistro.bistroapp.entity.Customer;
+import com.example.bistro.bistroapp.repository.CustomerRepository;
+import com.example.bistro.bistroapp.service.CustomerServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.*;
+import org.junit.jupiter.api.Assertions;
+
+public class CustomerServiceImplTest {
+
+    @Mock
+    private CustomerRepository customerRepository;
+
+    @InjectMocks
+    private CustomerServiceImpl customerService;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        ReflectionTestUtils.setField(customerService, "customerRepository", customerRepository);
+    }
+
+    @Test
+    public void testAddCustomer() {
+        Customer customer = new Customer();
```

Comment:
You should also check that customer has certain values for different fields like name etc.